learning_assistant/views.py

```diff
@@ -91,3 +91,41 @@ def post(self, request, course_id):
         status_code, message = get_chat_response(prompt_template, message_list, course_id)
 
         return Response(status=status_code, data=message)
+
+
+class LearningAssistantEnabledView(APIView):
+    """
+    View to retrieve whether the Learning Assistant is enabled for a course.
+
+    This endpoint returns a boolean representing whether the Learning Assistant feature is enabled in a course
+    represented by the course_key, which is provided in the URL.
+
+    Accepts: [GET]
+
+    Path: /learning_assistant/v1/course_id/{course_key}/enabled
+
+    Parameters:
+        * course_key: the ID of the course
+
+    Responses:
+        * 200: OK
```

Comment:
I'm guessing that since this is a GET which checks for enabled, we don't need to do any validation? For example, would we need to validate and return an 4** response if the course id doesn't exist?

Reply:
That's a good question. I don't think there's a spot in the code path where the existence of the course associated with the course ID is checked. [learning_assistant_enabled](https://github.com/edx/learning-assistant/blob/main/learning_assistant/api.py#L120) queries the `LearningAssistantCourseEnabled` model and handles the case where there is no instance for a given course ID. It also queries the course waffle flag, which shouldn't error if the course ID doesn't exist; it would just return `False` in that case.

I do think it would make sense to add a `try...except` around the `CourseKey.from_string(course_id)` line, in case the course ID is malformatted and can't be parsed into a `CourseKey`. I think the course pattern regular expression in the URL should take care of validating the course ID, but it doesn't hurt to be extra cautious here.

What do you think?